llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp

```diff
@@ -1370,9 +1370,9 @@ void AArch64AsmPrinter::emitFunctionEntryLabel() {
       MF->getFunction().getCallingConv() ==
           CallingConv::AArch64_SVE_VectorCall ||
       MF->getInfo<AArch64FunctionInfo>()->isSVECC()) {
-    auto *TS =
-        static_cast<AArch64TargetStreamer *>(OutStreamer->getTargetStreamer());
-    TS->emitDirectiveVariantPCS(CurrentFnSym);
+    if (auto *TS = static_cast<AArch64TargetStreamer *>(
```

Comment:
Looking at this again, I think we should only emit the variant PCS attribute when the target's triple is ELF (`TM.getTargetTriple().isOSBinFormatELF()`) because this is specifically an ELF attribute. This means you can remove this condition, because for ELF targets it should never return a nullptr.